File: 2019/day01/part2.js
const fs = require('fs');

const fuel = mass => {
    let f = Math.floor(mass / 3) - 2;
    return f > 0 ? f + fuel(f) : 0;
}

fs.readFile('input.txt', { encoding: 'utf-8' }, (err, data) => {
    if (err) throw err;
    const arr = data.split("\n");
    console.log(arr.reduce((a, b) => fuel(b) + a, 0))
});

File: 2019/day01/part2.py
import math


def fuel(mass):
    f = math.floor(mass / 3) - 2
    return f + fuel(f) if f > 0 else 0
    # return f > 0 ? f + fuel(f): 0


with open('input.txt') as fp:
    line = fp.readline()
    sum = 0
    while line:
        sum += fuel(int(line))
        line = fp.readline()
    print(sum)
